controllers/grid/catalogEntry/PublicationFormatCategoryGridDataProvider.inc.php

```diff
@@ -94,8 +94,7 @@ public function getRequestArgs()
     public function loadData($filter = [])
     {
         return Application::getRepresentationDAO()
-            ->getByPublicationId($this->getPublication()->getId())
-            ->toAssociativeArray();
+            ->getByPublicationId($this->getPublication()->getId());
```

Comment:
I *think* this is where you'll run into weird problems if you're not yielding these by ID.